doc/dart_documentation_comment_specification.md

```diff
@@ -0,0 +1,485 @@
+
+### **1.2. Scope**
+
+**In Scope**
+
+This specification covers the following topics:
+
+* The *syntax* for writing documentation comments.
+* The rules governing where documentation comments can be *placed*.
+* The mechanism for *resolving* references between elements.
+* The logic for *associating* documentation with code elements.
+
+**Out of Scope**
+
+This document does not cover:
+
+* The implementation details of specific documentation tools (e.g., `dartdoc`, `analysis server`).
+* The syntax and behavior of tool-specific directives (e.g., `@template`, `@canonicalFor`).
+* Any details related to the visual styling or HTML layout of generated documentation.
+* Best practices or style guides for writing effective documentation content.
+
+### **1.3. Terminology**
+
+* **Doc Comment:** A comment intended to be processed by documentation generation tools, like  dartdoc.
+* **Element**: Dart declarations or directives that can have documentation or be referenced in a doc comment: library directives, top-level, or member declarations such as class, menthod, or variable.
+* **Identifier:** An individual name in the code (e.g., `MyClass`, `myMethod`, `prefix`).
+* **Qualified Name:** A name composed of two or more *identifiers* separated by dots, used to access an element within a specific namespace (e.g., MyClass.myMethod, prefix.MyClass).
+* **Name:** A name is either a single *identifier* or a *qualified name*.
+* **Reference:** A string enclosed in square brackets within a doc comment (e.g., `[MyClass]` or `[MyClass.myMethod]`) that links to an element. The text inside the brackets is either a single *identifier* or a *qualified name*.
+
+## **2\. Syntax of Documentation Comments**
+
+### **2.1. Line-Based Doc Comments (Recommended standard)**
+
+* Starts with `///`.
+* All consecutive lines starting with `///` are treated as part of the same comment block.
+
+### **2.2. Block-Based Doc Comments (Historic, not recommended)**
+
+* Starts with `/**` and ends with the matching `*/`.
+* Block comments can be nested (e.g., `\*\* \* inner */ */`). Documentation tools must respect this, ensuring that the comment block only ends at the closing  \*/ that matches the opening delimiter.
+* The content within the delimiters is treated as the documentation.
+* Leading whitespace followed by a single asterisk and a single following space on subsequent lines are considered stylistic and are stripped by doc generators.
+
+**Example**
+
+```
+/**
+ *
+ *
+ */
+```
+
+### **2.3. Content Format (Markdown)**
+
+The text within a documentation comment block is parsed as CommonMark markdown, allowing for rich text formatting. This includes headings, lists, code blocks, and emphasis, which are converted for instance to HTML in the generated documentation.
```

Comment:
@szakarias in case you missed the other comments (I can't find them now, I think they got collapsed somewhere when this was updated), this is actually GFM rather than CommonMark.

Edit: https://github.com/dart-lang/dartdoc/pull/4130#discussion_r2577521545

Reply:
Ah great thanks, I was looking for the other comment since I could see that you had left one!